src/instance.rs

```diff
@@ -228,6 +228,17 @@ pub(crate) struct Py2Borrowed<'a, 'py, T>(
     Python<'py>,
 );
 
+impl<'py, T> Py2Borrowed<'_, 'py, T> {
+    /// Creates a new owned `Py2` from this borrowed reference by increasing the reference count.
+    pub(crate) fn to_owned(self) -> Py2<'py, T> {
```

Comment:
I took `self` by ownership here because it avoids the need for a closure in `.assume_borrowed(py).map(Py2Borrowed::to_owned)`.

Unsurprisingly clippy doesn't like this, and wants it to be called `into_owned`. What do you think of that?

Reply:
If `ToOwned/Borrow` were to support GAT, Clippy would also complain about this impl, wouldn't it? So I'd say Clippy is wrong and the lint should generally not fire for `Copy` types. Or maybe it does and we should just also add the `Clone` and `Copy` impls for `Py2Borrowed` here as well?

Reply:
Ah indeed for `Copy` types, clippy even wants `to_` methods to take `self`: https://rust-lang.github.io/rust-clippy/master/index.html#/wrong_self_convention

So I've cherry-picked the `Copy` commit from #3681 here.